src/main/groovy/com/athaydes/spockframework/report/internal/HtmlReportCreator.groovy

```diff
@@ -246,37 +257,85 @@ class HtmlReportCreator extends AbstractHtmlCreator<SpecData>
+                    writeBlockRow( builder, trCssClass( feature ),
+                            ( index == 0 ? block.kind : 'and' ), blockText )
                 }
-                writeBlockRow( builder, trCssClassArg,
-                        ( index == 0 ? block.kind : 'and' ), blockText )
+            else if ( !hideEmptyBlocks ) {
+                writeBlockRow( builder, trCssClass( feature ), block.kind, '----' )
+            }
+        }
+    }
+
+    private void writeBlocksFromCode( MarkupBuilder builder, FeatureInfo feature, IterationInfo iteration ) {
+        def blocks = codeReader.getBlocks( feature )
+        for ( BlockCode block in blocks ) {
+            def text = iteration && block.text ?
+                    stringProcessor.process( block.text, feature.dataVariables, iteration ) :
+                    ( block.text ?: '' )
+            def blockKind = block.label ?: 'Block:'
+            if ( text ) {
+                writeBlockRowFromCodeAndText( builder, trCssClass( feature ), blockKind, block.statements, text )
+            } else  {
+                writeBlockRowFromCodeOnly( builder, trCssClass( feature ), blockKind, block.statements )
             }
-        else if ( !hideEmptyBlocks )
-            writeBlockRow( builder, trCssClassArg, block.kind, '----' )
+        }
+    }
+
+    private trCssClass( FeatureInfo feature ) {
+        feature.skipped ? [ 'class': 'ignored' ] : null
     }
 
     private writeBlockRow( MarkupBuilder builder, cssClass, blockKind, text ) {
         builder.tr( cssClass ) {
             writeBlockKindTd( builder, blockKind )
-            td {
-                div( 'class': 'block-text', text )
-            }
+            writeBlockTextTd( builder, text )
+        }
+    }
+
+    private writeBlockRowFromCodeAndText( MarkupBuilder builder, cssClass, blockKind, List statements, text ) {
+        writeBlockRow ( builder, cssClass, blockKind, text )
+        if ( statements ) builder.tr {
+            td {}
+            writeCodeTd( builder, statements )
+        }
+    }
+
+    private writeBlockRowFromCodeOnly( MarkupBuilder builder, cssClass, blockKind, List statements ) {
+        if ( statements ) builder.tr( cssClass ) {
+            writeBlockKindTd( builder, blockKind )
+            writeCodeTd( builder, statements )
+        }
+    }
+
+    private void writeBlockKindTd( MarkupBuilder builder, blockKind ) {
+        builder.td {
+            div( 'class': 'block-kind', Utils.block2String[ blockKind ] )
+        }
+    }
+
+    private void writeBlockTextTd( MarkupBuilder builder, text ) {
+        builder.td {
+            div( 'class': 'block-text', text )
         }
     }
 
-    private void writeBlockKindTd( MarkupBuilder builder, blockKindKey ) {
+    private writeCodeTd( MarkupBuilder builder, List statements ) {
         builder.td {
-            div( 'class': 'block-kind', Utils.block2String[ blockKindKey ] )
+            pre( 'class': 'block-text', statements.join('\n') )
```

Comment:
This is not `block-text`. Block text has a specific meaning: the String documenting the block. This is something else.